bundles/binding/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/MyqData.java

```diff
@@ -82,16 +85,14 @@ public GarageDoorData getMyqData() {
 	 *            Attempt number when it recursively calls itself
 	 */
 	private String getGarageStatus(int attemps) {
-		if (this.sercurityTokin == null && Login()) {
+		if (sercurityTokin == null && !Login()) {
```

Comment:
Sounds good, I'm actually playing with this logic now.  I had two events that caused my token to invalidate and received different errors from the MyQ api, one was at least the expire case, which the logic was not handling well.  

Do you think we need a timer?  Can we just catch the error ( code -3333, weird ) and retry the call?  That way if they increase, decrease or do something else to the login token the code handles it correctly?  I have integrated with a few API's like this and found that you eventually have to handle the error code in any case, which makes the timer logic redundant.  I have a flight tomorrow so I'll have time to play, I'll try and commit what I'm thinking.